frontend/src/components/taskSelection/action.js

```diff
@@ -304,12 +344,40 @@ export function TaskMapAction({ project, projectIsReady, tasks, activeTasks, act
                   </>
                 )}
                 {activeSection === 'history' && (
-                  <TaskHistory
-                    projectId={project.projectId}
-                    taskId={tasksIds[0]}
-                    commentPayload={taskHistory}
-                    mapperLevel={userDetails['mappingLevel']}
-                  />
+                  <>
+                    {activeTasks.length === 1 && (
+                      <>
+                        <TaskHistory
+                          projectId={project.projectId}
+                          taskId={tasksIds[0]}
+                          commentPayload={taskHistory}
+                          mapperLevel={userDetails['mappingLevel']}
+                        />
+                      </>
+                    )}
+                    {action === 'VALIDATION' && activeTasks.length > 1 && (
+                      <Accordion allowMultipleExpanded allowZeroExpanded onChange={handleFetch}>
```

Comment:
We could move that component to a new file. That action.js is too big already. We can also write tests to it as it's isolated.